src/test/java/com/hocs/test/pages/managementUI/TeamManagement.java

```diff
@@ -0,0 +1,135 @@
+package com.hocs.test.pages.managementUI;
+
+import static net.serenitybdd.core.Serenity.sessionVariableCalled;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import com.hocs.test.pages.Page;
+import cucumber.api.java.en.When;
+import net.serenitybdd.core.annotations.findby.FindBy;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+
+import static net.serenitybdd.core.Serenity.setSessionVariable;
+
+public class TeamManagement extends Page {
+
+    @FindBy(xpath = "//input[@id='Teams-input']")
+    public WebElementFacade teamSearchBar;
+
+    @FindBy(xpath = "//button[@class='govuk-button view-team-button']")
+    public WebElementFacade viewTeamButton;
+
+    @FindBy(xpath = "//button[@class='govuk-button add-team-members-button']")
+    public WebElementFacade addTeamMembersButton;
+
+    @FindBy(xpath = "//input[@id='users-input']")
+    public WebElementFacade userSearchBar;
+
+    @FindBy(xpath = "//button[@class='govuk-button view-team-button']")
+    public WebElementFacade addSelectedUsersButton;
+
+    @FindBy(xpath = "//h2[@class='govuk-heading-l']")
+    public WebElementFacade teamNameHeader;
+
+    @FindBy(xpath = "(//td//a)[1]")
+    public WebElementFacade firstRemoveButtonInList;
+
+    @FindBy(xpath = "(//div//p)[2]")
+    public WebElementFacade errorMessage;
+
+    public void assertTeamManagementPageTitle() {
+        assertThat($("//h1").getText(), is("Team search"));
+    }
+
+    public void selectATeam(String teamName) {
+        waitABit(500);
+        typeInto(teamSearchBar, teamName);
+        setSessionVariable("teamName").to(teamName);
+        teamSearchBar.sendKeys(Keys.ENTER);
+        clickOn(viewTeamButton);
+    }
+
+    public void selectTeamWithNoUsers() {
+        waitABit(500);
+        typeInto(teamSearchBar, "Data & Identity Unit");
+        teamSearchBar.sendKeys(Keys.ENTER);
+        clickOn(viewTeamButton);
+    }
+
+    public void selectAUser(String nameOfUser) {
+        clickOn(addTeamMembersButton);
+        typeInto(userSearchBar, nameOfUser);
+        setSessionVariable("nameOfUser").to(nameOfUser);
+        waitABit(4000);
+        userSearchBar.sendKeys(Keys.ENTER);
+        clickOn(addSelectedUsersButton);
+    }
+
+    public void clickAddSelectedUsers() {
+        clickOn(addTeamMembersButton);
+        clickOn(addSelectedUsersButton);
+    }
+
+    public void removeUserFromTeamWithAssignedCases() {
+        String nameOfUser = sessionVariableCalled("nameOfUserWithCases").toString();
+        WebElementFacade removeButtonOfUser = findAll("//td[@class='govuk-table__cell'][contains(text(), '" + nameOfUser +
```

Comment:
Wouldnt Xpath selector ("//td[@class='govuk-table__cell'][contains(text(), 'Danny Large')]/following-sibling::td") work here without need to get array of elements and narrow to first?

Reply:
Probably right. I just based this off similar code in the pack and it worked for the test.